src/components/List/src/bem-mapping.ts

```diff
@@ -0,0 +1,32 @@
+export const list_classes = {
+  root: 'denhaag-list',
+};
+
+export const listitemicon_classes = {
+  root: 'denhaag-list__itemicon',
+};
+
+export const listsubheader_classes = {
+  root: 'denhaag-list__subheader',
+};
+
+export const listitem_classes = {
+  root: 'denhaag-list__item',
+  focusVisible: 'denhaag-list__itemfocused',
+};
+
+export const listitemtext_classes = {
+  root: 'denhaag-list__itemtext',
+};
+
+export const listitemsecondaryaction_classes = {
+  root: 'denhaag-list__secondaryaction',
+};
+
+export const listitemtextprimary_classes = {
+  root: 'denhaag-list__itemtextprimary',
```

Comment:
We're getting into some rather complex subcomponents here, so the BEM naming also gets a bit complicated here. 
In this case it should at least be `item-text` instead of `itemtext`. But furthermore, are these two different elements, or are these basically the same element with a modifier applied to it?

In the first case I would argue we should use `denhaag-list__item-primary-text` and `denhaag-list__item-secondary-text`. That way `primary` and `secondary` don't look like a modifier with improperly applied BEM naming and we use a sort of natural language flow and remain consistent with the naming of for example `denhaag-list__item-secondary-action`.

If we decide these are actually both just the element (for example) `denhaag-list__item-text-content` and the secondary text just has some different styling than the primary text we would get the following classes: `denhaag-list__item-text-content` and `denhaag-list__item-text-content--secondary`. Note that a modifier class is always a class you add, you do not use it to replace the original class. This would mean for our secondary text we would get `root: denhaag-list__item-text-content denhaag-list__item-text-content--secondary`. 
Another option would be to do the following:

```
export const listitemtextcontent_classes = {
   root: 'denhaag-list__item-text-content',
   body2: 'denhaag-list__item-text-content--secondary'
}
```

And apply `variant="body2"` to the Typography component, in which case the MUI component would automatically add the correct class, but this implementation is highly dependent on our implementation of the Typography component so I would at this moment stay away from this.

Reply:
I agree the subcomponents become a bit more complex here.
The `denhaag-list__item-text`  is applied to the react component (div) that is the parent to two distinct header elements `denhaag-list__item-text-primary` and `-secondary`, where secondary is optional.
Something similar to this:
```
<div class="denhaag-list__item-text">
  <h class="list__item-text-primary"></h>
  <h class="list__item-text-secondary"></h>
<div>
```
Since primary and secondary are not different styles applied to the same element, I don't think we should use a --flag here.

I will do some renaming though to make it more readable and consistent.